CMakeLists.txt

```diff
@@ -117,23 +124,22 @@ endif()
 
 include_directories("${JAVA_HBASE_DIR}/hbase-common/target/generated-sources/native/")
 ############
-## Validate that we have C++ 14 support
+## Validate that we have C++ 17 support
```

Comment:
Yes my hope was to move forward as it would support optional. In retrospect after a nice break from computers and a week off I don't want to push for this change without a lot more testing. I will revert this part of the change. I've grown fond of c++17, but I think you'd agree it's an unnecessary change at this time. Thanks for your patience and help!

Reply:
Ya, preferred to keep at 14 unless there is a lot of upside IMO. Like I mentioned in my other patch, we are going to have a precommit soon, so we will have good test coverage in a clean environment once we get #10 in.